### ISSUE TITLE
recorded / post processed audio opus corrupted

### ISSUE THREAD

**Issue by tarasian666** [meetecho/janus-gateway#2292]
#2290
Audio file often is like cut out in the middle (maybe silence is cutted out). 
Difference in time with video duration can be huge (bigger than difference in created/written time of files)
IDK if it is post process issue or in recording mjr.
Yes, audio and video is with different length but difference should be equal to difference in start time and not 5 min as it is in this example:
```
janus-pp-rec videoroom-1813116376865619-user-7102993389991015-1594222623284314-video.mjr video.webm
Janus version: 102 (0.10.2)
Janus commit: not-a-git-repo
Compiled on:  Thu Jul  9 16:39:58 UTC 2020

Logging level: 4

Source file: videoroom-1813116376865619-user-7102993389991015-1594222623284314-video.mjr
Target file: video.webm

File is 74260379 bytes
Pre-parsing file to generate ordered index...
This is a video recording:
  -- Codec:   vp8
  -- Created: 1594222623284336
  -- Written: 1594222623300242
SSRC detected: 401544988
Counted 68912 RTP packets
Counted 68912 frame packets
  -- 640x480 (fps [5940,103410] ~ 15)
[libvpx @ 0x561b454cbb00] v1.8.2
[libvpx @ 0x561b454cbb00] Neither bitrate nor constrained quality specified, using default CRF of 32 and bitrate of 256kbit/sec
First keyframe: 15030
video.webm is 72029444 bytes
Bye!
```
```
janus-pp-rec videoroom-1813116376865619-user-7102993389991015-1594222623284314-audio.mjr audio.opus
Janus version: 102 (0.10.2)
Janus commit: not-a-git-repo
Compiled on:  Thu Jul  9 16:39:58 UTC 2020

Logging level: 4

Source file: videoroom-1813116376865619-user-7102993389991015-1594222623284314-audio.mjr
Target file: audio.opus

File is 5112848 bytes
Pre-parsing file to generate ordered index...
This is an audio recording:
  -- Codec:   opus
  -- Created: 1594222623284315
  -- Written: 1594222623287371
SSRC detected: 2385931017
Counted 51893 RTP packets
Counted 51893 frame packets
Writing .opus file header
audio.opus is 5009186 bytes
Bye!
```

```
ffprobe -hide_banner video.webm 
[vp8 @ 0x55a9f29a8dc0] Invalid profile 7.
Input #0, matroska,webm, from 'video.webm':
  Metadata:
    COMMENT         : {"t": "v", "c": "vp8", "s": 1594222623284336, "u": 1594222623300242}
    ENCODER         : Lavf58.45.100
  Duration: 00:21:21.96, start: 0.000000, bitrate: 449 kb/s
    Stream #0:0: Video: vp8, yuv420p(progressive), 640x480, SAR 1:1 DAR 4:3, 29.97 tbr, 1k tbn, 1k tbc (default)
    Metadata:
      DURATION        : 00:21:21.958000000
```
```
ffprobe -hide_banner audio.opus 
Input #0, ogg, from 'audio.opus':
  Duration: 00:17:17.86, start: 0.000000, bitrate: 38 kb/s
    Stream #0:0: Audio: opus, 48000 Hz, stereo, fltp
    Metadata:
      comment         : {"t": "a", "c": "opus", "s": 1594222623284315, "u": 1594222623287371}

```

**Comment by ikakey** [comment]
@lminiero 

please read carefully what @tarasian666 tarasias666 wrote, and do not rush to close the issue 

thank you!

**Comment by atoppi** [comment]
The audio file looks correct.
51893 RTP opus packets are equal to ~ 17 mins and 18 secs considering a 20 ms packet length.

**Comment by tarasian666** [comment]
> The audio file looks correct.
> 51893 RTP opus packets are equal to ~ 17 mins and 18 secs considering a 20 ms packet length.

And video? If all ok in mjr files, so problem is in recording process?

The final result of these two files:
At beginning all is in sync, after few minutes user went to do some stuff and left turned all on (video and audio). So there was video of empty chair and silent audio and while user was still away audio jumps from end of the conversation to the middle, after user returned "in frame" there was only video of this user moving his lips...

**Comment by atoppi** [comment]
It is very unlikely that Janus did not record some packets in the middle. The only reason I may think of is that those packets are getting discarded for some reason.
Given the description you are providing, it sounds like something related to the device on the user end.

Can you share the mjr files? We could analyze them to see if the RTP stream has got any issue.

**Comment by tarasian666** [comment]
> It is very unlikely that Janus did not record some packets in the middle. The only reason I may think of is that those packets are getting discarded for some reason.
> Given the description you are providing, it sounds like something related to the device on the user end.
> 
> Can you share the mjr files? We could analyze them to see if the RTP stream has got any issue.

https://file.io/NyCa73UmBFI4

**Comment by tarasian666** [comment]
@atoppi did you downloaded files or i need to reupload?

**Comment by lminiero** [comment]
It's not like we don't have anything to do. We'll get back to you when we have time to do that.

**Comment by ikakey** [comment]
> It's not like we don't have anything to do. We'll get back to you when we have time to do that.

Hi @lminiero, it was very simple question why are you so angry ? you did not even read the previous issue and immediately closed it, now it turns out that yes, there is a problem, and instead of answering a simple question about whether you received a file that confirms the bug, and helps you and us to resolve it, you answer the question with anger and disrespect.
I understand that we are all tense because of the situation in the world, but we need to try to remain human, this is what makes us here, on the github, a community that by joint efforts is trying to change the world for the better.  I hope for your understanding and wish you good health!

**Comment by lminiero** [comment]
Because I'm tired of people expecting me to run to fix issues the moment they're opened, as if I work for them. We're busy: when we have time, we'll look into this (IF there is a problem to fix, which has yet to be confirmed).

**Comment by atoppi** [comment]
@tarasian666 that url returns 404.

**Comment by ikakey** [comment]
> Because I'm tired of people expecting me to run to fix issues the moment they're opened, as if I work for them. We're busy: when we have time, we'll look into this (IF there is a problem to fix, which has yet to be confirmed).

No one ask you run issue immediately, these was a question about file :) 

ok, let’s relax everyone and continue do what we need to do, step by step 🙏

**Comment by lminiero** [comment]
A file link with a very short expiration date implies I have to download and look at it right away. That feels like rushing me. If you put it somewhere where we can download it on our own time, we'll have a look at it when we can.

**Comment by ikakey** [comment]
> A file link with a very short expiration date implies I have to download and look at it right away. That feels like rushing me. If you put it somewhere where we can download it on our own time, we'll have a look at it when we can.

you right, I’m sorry 🙏

**Comment by tarasian666** [comment]
> > https://file.io/NyCa73UmBFI4
> 
> @tarasian666 that url returns 404.

sent shared link to your email

**Comment by atoppi** [comment]
The audio packet with seq num **20503** has been received by Janus **244** seconds after the previous one but the RTP timestamp of that packet has not been increased accordingly (incremented of 960 units that is **20 ms** in this case).
`pp-rec` will treat this situation as a huge delay in the packet delivery (because this is what it looks like from the receiver perspective). The final track length will follow the available RTP data and since no losses, no seq num and no timestamp jumps are present in the RTP stream, the audio track will be exactly ~ 17 mins and 18 secs.

In conclusion it seems that something has "paused" the media delivery for 244 seconds (**that roughly equals to the 4 minutes difference you have between audio and video**).
My guess is that this behavior is related to the sender endpoint or network (to double check try to get a pcap on that side too).

In order to confirm all of this by yourself check the `mjr2pcap` tool that is part of Janus source.

**Comment by atoppi** [comment]
Given the lack of feedback, I'm assuming this issue can be closed.
